### PR TITLE
8269148: Update minor GCC version in GitHub Actions pipeline

### PR DESCRIPTION
**Pull request by shipilev**
It seems Ubuntu had bumped the version for GCC, so GHA started to fail with e.g.:

```
The following packages have unmet dependencies:
 g++-10-s390x-linux-gnu : Depends: gcc-10-s390x-linux-gnu-base (= 10.2.0-5ubuntu1~20.04cross1) but 10.3.0-1ubuntu1~20.04cross1 is to be installed
 gcc-10-s390x-linux-gnu : Depends: cpp-10-s390x-linux-gnu (= 10.2.0-5ubuntu1~20.04cross1) but 10.3.0-1ubuntu1~20.04cross1 is to be installed
                          Depends: gcc-10-s390x-linux-gnu-base (= 10.2.0-5ubuntu1~20.04cross1) but 10.3.0-1ubuntu1~20.04cross1 is to be installed
E: Unable to correct problems, you have held broken packages.
Error: Process completed with exit code 100.
```

I believe we should just update to `10.3.0-1ubuntu1~20.04`.

Additional testing:
 - [x] GitHub Actions, package installation steps work
 - [x] GitHub Actions, the builds complete

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8269148](https://bugs.openjdk.java.net/browse/JDK-8269148): Update minor GCC version in GitHub Actions pipeline

### Reviewers
 * [Erik Joelsson](https://openjdk.java.net/census#erikj) (@erikj79 - **Reviewer**)
 * [David Holmes](https://openjdk.java.net/census#dholmes) (@dholmes-ora - **Reviewer**)
 * [Xin Liu](https://openjdk.java.net/census#xliu) (@navyxliu - no project role)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk17 pull/120/head:pull/120` \
`$ git checkout pull/120`

Update a local copy of the PR: \
`$ git checkout pull/120` \
`$ git pull https://git.openjdk.java.net/jdk17 pull/120/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 120`

View PR using the GUI difftool: \
`$ git pr show -t 120`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk17/pull/120.diff">https://git.openjdk.java.net/jdk17/pull/120.diff</a>

</details>

### CODE REVIEW

**Review by erikj79 — APPROVED** [review]

File: .github/workflows/submit.yml [hunk 1]
```diff
@@ -172,7 +172,7 @@ jobs:
       - name: Install dependencies
         run: |
           sudo apt-get update
-          sudo apt-get install gcc-10=10.2.0-5ubuntu1~20.04 g++-10=10.2.0-5ubuntu1~20.04 libxrandr-dev libxtst-dev libcups2-dev libasound2-dev
+          sudo apt-get install gcc-10=10.3.0-1ubuntu1~20.04 g++-10=10.3.0-1ubuntu1~20.04 libxrandr-dev libxtst-dev libcups2-dev libasound2-dev
```
Comment (navyxliu) [inline .github/workflows/submit.yml:175]: One dummy question: 
Why do we need to specify the real package name here? 
If weinstall gcc-10, I think apt system will pick up the latest gcc-10 for you.

**Review by dholmes-ora — APPROVED** [review]

**Comment by navyxliu** [comment]
Thanks for the explanation. The reason I ask because I guess we need to review every minor gcc updates in the future. 
For a sizeable c/c++ project, it indeed needs extra caution to bump tool-chains. 
LGTM.

**Review by navyxliu — APPROVED** [review]

**Comment by dholmes-ora** [comment]
There is a new crash in a vector compiler test that we have not seen before in any of our testing. Could it be related to the change in gcc version?

**Comment by navyxliu** [comment]
This PR doesn't touch anything for linux_x86, but the crash happened on linux-x86 platform as well. 
Therefore, it's very unlikely to be caused by toolchains change.

I think it's JDK-8269179

**Comment by shipilev** [comment]
I am integrating this to unbreak the testing.

/integrate

**Pull request closed without merging**